testing/adios2/engine/staging-common/CMakeLists.txt

```diff
@@ -202,13 +202,32 @@ endif()
 if (NOT MSVC)    # not on windows
     # Local Vars don't work for InSitu
     set (BP_TESTS ${ALL_SIMPLE_TESTS})
+    # Delayed reader not worth testing on file engines
+    list (FILTER BP_TESTS EXCLUDE REGEX "DelayedReader")
+    # SharedVars fail with BP*
     list (FILTER BP_TESTS EXCLUDE REGEX ".*SharedVar$")
     foreach(test ${BP_TESTS})
         add_common_test(${test} BP3)
         add_common_test(${test} BP4)
     endforeach()
 endif(NOT MSVC)
 
+#
+#   Setup tests for HDF5 engine
+#
+if (NOT MSVC)    # not on windows
+    set (HDF5_TESTS ${ALL_SIMPLE_TESTS})
+    # Delayed reader not worth testing on file engines
+    list (FILTER HDF5_TESTS EXCLUDE REGEX "DelayedReader")
+    # Local variables don't work on hdf5 
+    list (FILTER HDF5_TESTS EXCLUDE REGEX ".*Local")
+    # Oddly, some multi-writer tests fail
+    list (FILTER HDF5_TESTS EXCLUDE REGEX "[2345]x")
+    foreach(test ${HDF5_TESTS})
```

Comment:
Just for future reference, i'm trying to move us to using the newer syntax of `foreach`:
```cmake
# Old
foreach(var ${var_list_name})
  #do things
endforeach()

# New
foreach(far IN LISTS var_list_name)
  # do stuff
endforeach()
```

It's not worth updating this pr for but it's generally good cmake practice to use the newer syntax.

Reply:
I will take note.  (I didn't know there was a newer syntax...)

Reply:
In general it's one of the many changes made over the years to try to work on variables directly without having to dereference them.  The newer syntax addresses issues that arise when items in the list might have spaces in them and end up evaluated as separate items.